types/express-serve-static-core/index.d.ts

```diff
@@ -447,7 +460,7 @@ export interface Request<P extends Params = ParamsDictionary> extends http.Incom
     xhr: boolean;
 
     //body: { username: string; password: string; remember: boolean; title: string; };
-    body: any;
+    body: { [key: string]: string | undefined };
```

Comment:
@micksatana , As you wrote, by default (before using any body parsing middleware), req.body is undefined, exactly what my PR intends to resolve.

If we do not have that middleware, there will be no `body` property, so what `body: any;` is doing is the equivalent of `undefined: any;`

Right now, `body: any;` is saying is that the request object is going to have a `body` property 100% of the time and its type is `any`. How is that possible? If we are not using that middleware, then there should not be a `body` property of defined type of `any`, because we cannot assign a type of `any` to `undefined` and my proof is the fact that TypeScript does not give us a warning if we omit the use of `bodyParser` middleware.

So if there is no `body` property, then doing something like this `const { email, password } = req.body;` is going to throw an error at me.  We want TypeScript to warn us about that error.

So my PR is helping TypeScript understand that `req.body` is going to contain an object that might or might not have the properties the developer expects. A type of `any` is not helpful, `any` does not let TypeScript know that `req.body` may be `undefined`, an object stating that we do not know what the keys will be inside of this object does. So its saying this object is going to have some keys that are strings `{ [key: string] }`, we don't know what they are, but they are definitely going to exist. As a value for all these keys I put in `string | undefined`. Do you see that there, its explicitly saying it may be `string` or it may be `undefined` as the Express documentation says.

Reply:
> Right now, body: any; is saying is that the request object is going to have a body property 100% of the time and its type is any.

I believe that is correct we expect `body` property to exists and it could be `undefined` but not undeclared.

> So if there is no body property, then doing something like this const { email, password } = req.body; is going to throw an error at me. We want TypeScript to warn us about that error.

Could you write a test case for this in `express-serve-static-core-tests.ts`? And try switching between `any` and the new type `{ [key: string]: string | undefined }`. I do not see any error shown in both cases. The type `{ [key: string]: string | undefined }` will cause error in case `req.body.someVar` is any other types except `string`; for example, `number`. This will definitely cause error in other apps using this module.

> its explicitly saying it may be string or it may be undefined as the Express documentation says.

I might miss the section "it may be string or it may be undefined". Please give the reference URL and where to look. From what I see they only mentioned...

> **req.body**
> Contains key-value pairs of data submitted in the request body. By default, it is undefined, and is populated when you use body-parsing middleware such as express.json() or express.urlencoded().

By key-value means, key is `string` and value can be anything. But by default, the `req.body` is `undefined` type, not empty object or `{ [key: string]: undefined }` type.

Reply:
You might want to discuss in #39400 as well because if the PR passed this may be in conflict.